### PR TITLE
nixos/misc: add VARIANT_ID for enriching /etc/os-release

### PR DESCRIPTION
**Pull request by Lassulus**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/profiles/installation-device.nix [hunk 1]
```diff
@@ -20,6 +20,7 @@ with lib;
     ];
 
   config = {
+    system.nixos.variant_id = "installer";
```
Comment (RaitoBezarius) [inline nixos/modules/profiles/installation-device.nix:23]: mkDefault?

Comment (Mic92) [inline nixos/modules/profiles/installation-device.nix:23]:
```suggestion
    system.nixos.variant_id = lib.mkDefault "installer";
```

Comment (Mic92) [reply nixos/modules/profiles/installation-device.nix]: Why not.

File: nixos/modules/misc/version.nix [hunk 2]
```diff
@@ -28,6 +28,8 @@ let
     DOCUMENTATION_URL = "https://nixos.org/learn.html";
     SUPPORT_URL = "https://nixos.org/community.html";
     BUG_REPORT_URL = "https://github.com/NixOS/nixpkgs/issues";
+  } // lib.optionalAttrs (cfg.variant_id == null) {
```
Comment (R-VdP) [inline nixos/modules/misc/version.nix:31]: Should we not invert the condition here? We want to include it if `cfg.variant_id != null`, right?

Comment (Mic92) [reply nixos/modules/misc/version.nix:31]: :)

Comment (Mic92) [inline nixos/modules/misc/version.nix:31]:
```suggestion
  } // lib.optionalAttrs (cfg.variant_id != null) {
```

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**